warehouse/core/src/main/java/datawave/iterators/filter/ageoff/DataTypeParser.java

```diff
@@ -0,0 +1,154 @@
+package datawave.iterators.filter.ageoff;
+
+import org.apache.accumulo.core.data.ArrayByteSequence;
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+
+/**
+ * Originally extracted from DataTypeAgeOffFilter
+ *
+ * Contains a single public static method for parsing the datatype (as a ByteSequence) from a key.
+ *
+ * It can parse the various shard schema key structures including the d, fi, tf, and event schemas.
+ *
+ * It can also parse an index table key.
+ */
+public class DataTypeParser {
+    
+    private static final char FI_CF_CHAR_1 = 'f';
+    private static final char FI_CF_CHAR_2 = 'i';
+    
+    private static final char TF_CF_CHAR_2 = 'f';
+    private static final char TF_CF_CHAR_1 = 't';
+    
+    private static final char DOC_CF_CHAR = 'd';
+    
+    private static final int NULL_BYTE = 0x00;
+    
+    // shard ID's minimum length = sum of the minimum length of its components: year + month + day + underscore + single digit partition
+    private static final int MIN_LENGTH_SHARD_ID = 10;
+    
+    /**
+     * @param key
+     *            Accumulo key to parse
+     * @param isIndexTable
+     *            true if the key is from an index table, false if the shard table
+     * @return a ByteSequence containing the data type name, possibly returning null or empty for unexpected column data.
+     */
+    public static final ByteSequence parseKey(Key key, boolean isIndexTable) {
+        /**
+         * Supports the shard and index table. There should not be a failure, however if either one is used on the incorrect table
```

Comment:
Help me understand this comment.  What happens if either one is used on the incorrect table?  "There should not be a failure, however if either one is used on the incorrect table" ________?

Reply:
Thanks for pointing this out.  It's a comment from the old code that was extracted to this class.
https://github.com/NationalSecurityAgency/datawave/pull/1729/files#diff-8ae64ed5c5daf1aac02cb892e5fafbb8fdbdcc3498a8fff042f19465eafec5ebL126

I'll push changes to reword these old comments.